src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs

```diff
@@ -139,6 +139,25 @@ public override Array GetEnumValues()
             return ret;
         }
 
+        public override Array GetEnumValuesAsUnderlyingType()
+        {
+            if (!IsActualEnum)
+                throw new ArgumentException(SR.Arg_MustBeEnum, "enumType");
+
+            // Get all of the values
+            ulong[] values = Enum.InternalGetValues(this);
+
+            Array ret = Array.CreateInstance(Enum.InternalGetUnderlyingType(this), values.Length);
+
+            for (int i = 0; i < values.Length; i++)
+            {
+                object val = Enum.ToObject(this, values[i]);
+                ret.SetValue(val, i);
+            }
```

Comment:
Would it be worthwhile special-casing each underlying type, in order to avoid all of the boxing and going through the slow SetValue path and using Array.CreateInstance and so on? e.g.
```C#
switch (Type.GetTypeCode(Enum.InternalGetUnderlyingType(this).GetTypeCodeImpl())
{
    case TypeCode.Byte:
    {
        byte[] ret = new byte[values.Length];
        for (int i = 0; i < values.Length; i++)
        {
            ret[i] = (byte)values[i];
        }
        return ret;
    }

    case TypeCode.UInt32:
    {
        uint[] ret = new uint[values.Length];
        for (int i = 0; i < values.Length; i++)
        {
            ret[i] = (uint)values[i];
        }
        return ret;
    }
    
    ... // etc.
}
```
?